qadence/finitediff.py

```diff
@@ -0,0 +1,47 @@
+from __future__ import annotations
+
+from typing import Callable
+
+import torch
+from torch import Tensor
+
+
+def finitediff(
+    f: Callable,
+    x: Tensor,
+    derivative_indices: tuple[int, ...],
+    eps: float = None,
+) -> Tensor:
+    """
+    Arguments:
+
+        f: Function to differentiate
+        x: Input of shape `(batch_size, input_size)`
+        derivative_indices: which *input* to differentiate (i.e. which variable x[:,i])
+        eps: finite difference spacing
+    """
+
+    if eps is None:
+        order = len(derivative_indices)
+        # FIXME: this is the float64 machine epsilon, should probably be fixed for float32
+        eps = 2.220446049250313e-16 ** (1 / (2 + order))
```

Comment:
@nmheim Isn't there a constant available somewhere in the Python standard library for machine precision?